geoh5py/workspace/workspace.py

```diff
@@ -1230,35 +1214,23 @@ def register(self, entity: Entity | EntityType | PropertyGroup):
 
         :param entity: The entity to be registered.
         """
+        dict_entity: dict[uuid.UUID, ReferenceType]
         if isinstance(entity, EntityType):
-            self._register_type(entity)
+            dict_entity = self._types
         elif isinstance(entity, Group):
-            self._register_group(entity)
+            dict_entity = self._groups
         elif isinstance(entity, Data):
-            self._register_data(entity)
+            dict_entity = self._data
         elif isinstance(entity, ObjectBase):
-            self._register_object(entity)
+            dict_entity = self._objects
         elif isinstance(entity, PropertyGroup):
-            self.register_property_group(entity)
+            dict_entity = self._property_groups
+            if not entity.on_file:
+                self.add_or_update_property_group(entity)
```

Comment:
**question:** no consequence of calling `self.add_or_update_property_group(entity)` before registering it?
It used to register first.